### PR TITLE
PerfMap: Log information about both ngened and jited methods.

### PR DESCRIPTION
**Pull request by kbaladurin**
This change provide ability for perf to symbolize backtraces with methods from NI files.

### CODE REVIEW

**Comment by kbaladurin** [comment]
@iarischenko @sergign60 @alpencolt

**Comment by jkotas** [comment]
Have you looked at `CreatePerfMap` option for crossgen?

It was our strategy to generate the perfmap for native images ahead of time - both for correctness (the prestub is not guaranteed to execute for all methods in native images) and performance (the method size lookup you are adding is not free).

File: src/vm/prestub.cpp [hunk 1]
```diff
@@ -261,6 +261,12 @@ PCODE MethodDesc::PrepareInitialCode()
     NotifyGdb::MethodPrepared(this);
 #endif
 
+#ifdef FEATURE_PERFMAP
+    EECodeInfo codeInfo(pCode);
+    size_t sizeOfCode = codeInfo.GetCodeManager()->GetFunctionSize(codeInfo.GetGCInfoToken());
```
Comment (jkotas) [inline src/vm/prestub.cpp]: This lookup is relatively expensive. It should be done only if we actually need to log the method size.

**Comment by kbaladurin** [comment]
@jkotas thank you! Yes, I looked at `CreatePerfMap` option for crossgen. In the case of separated perfmap for crossgen, it's necessary to recalculate symbol addresses and then merge all maps together. I think it's better to get one perfmap for all symbols if it's possible.

In what cases isn't prestub used? What is the alternative way to call such methods? Is there a common place where we can log method info for both ngened and jited methods?

**Comment by kbaladurin** [comment]
Thank you!

**Comment by brianrob** [comment]
@kbaladurin, thanks for the contribution.  This is actually something that I've tried to do in the past, but have been unable to make it work.  The reason is that the symbol resolution code in perf_events knows that even though this code isn't in an ELF formatted image, it also isn't JIT-compiled because it's not in private memory - it is in a memory mapped file with execute permissions.  Thus, it assumes that it is dealing with unknown pre-compiled code, and not JIT-compiled code and doesn't query /tmp/perf-<pid>.map.  Has this changed in newer versions of perf?

**Pull request closed without merging**